src/Ersatz/Orderable.hs

```diff
@@ -57,6 +66,44 @@ instance Orderable Bit where
   a <?  b = not a && b
   a <=? b = not a || b
 
+-- | Compare by lexicographic order on sorted key-value pairs
+instance (Ord k, Orderable v) => Orderable (Map.Map k v) where
+  x <?  y = assocsLt (Map.assocs x) (Map.assocs y)
+  x <=? y = assocsLe (Map.assocs x) (Map.assocs y)
+
+-- | Compare by lexicographic order on sorted key-value pairs
+instance Orderable v => Orderable (IntMap.IntMap v) where
+  x <?  y = assocsLt (IntMap.assocs x) (IntMap.assocs y)
+  x <=? y = assocsLe (IntMap.assocs x) (IntMap.assocs y)
+
+assocsLt :: (Ord k, Orderable v) => [(k,v)] -> [(k,v)] -> Bit
```

Comment:
Perhaps a silly question, but what is the advantage of doing it this way rather than relying on the `Orderable` instance for lists and pairs? That is, why have a `(Ord k, Orderable v)` context rather than a `(Orderable k, Orderable v)` context?

Reply:
`Orderable` exists for things that might contain `Bit`. Map keys never contain `Bit` because `Bit` aren't `Ord` (but the Map keys are definitely Ord). There's a similar thing going on in the Equatable instance where we use Eq on the keys first to avoid emitting any symbolic bits at all if the keys don't match.

For `IntMap` it matters a little less, but it means that we avoid emitting symbolic boolean expressions when comparing the Int keys.

Reply:
And while we're talking about it, it would be lovely if our Generics derived instances were able to know they were working with a type that had no Bit values possible and was able to switch over to `Eq` and `Ord` as an optimization, but I have no idea how to achieve that without making a mess of the API

Reply:
Ah, I had overlooked the fact that the `Equatable` instances for maps use the same convention. In that case, please ignore this suggestion.